### PR TITLE
Add recommendation controller and service tests

### PR DESCRIPTION
**Pull request by phlpphcknr**
- add missing tests
- formatting in globalstyles.js

### CODE REVIEW

File: backend/src/test/java/com/hackner/musicompass/service/RecommendationServiceTest.java [hunk 1]
```diff
@@ -141,7 +142,98 @@ public void setRecommendationTags(){
...
+                .gender(gender)
+                .genres(genres)
+                .roles(roles).build();
+        when(artistMongoDb.findAll()).thenReturn(getArtistList());
+
+        //WHEN
+        String actual = recommendationService.getArtistRecommendation(recommendationRequestDto);
+
+        //THEN
+        assertThat(actual, is("error-not-found"));
```
Comment (jamarob) [inline backend/src/test/java/com/hackner/musicompass/service/RecommendationServiceTest.java:163]: Do not use special strings for status! This is really bad and just asking for trouble.

File: frontend/src/components/GlobalStyle.js [hunk 2]
```diff
@@ -11,33 +11,34 @@ export default createGlobalStyle`
     background-color: var(--seondary-color);
     --primary-color: #fdcb6e;
     --secondary-color: moccasin;
-    --opacity-hover:0.7;
+    --opacity-hover: 0.7;
   }
-  
-  section{
+
+  section {
```
Comment (jamarob) [inline frontend/src/components/GlobalStyle.js:17]: not necessary, please delete

**Review by jamarob — CHANGES_REQUESTED** [review]
💯 for more tests, but that error string is really bad practice 😢 hence no approal from me.

File: backend/src/test/java/com/hackner/musicompass/controller/RecommendationControllerTest.java [hunk 3]
```diff
@@ -111,7 +113,64 @@ public void changeExistingRecommendationTags(){
...
+        ResponseEntity <String> response = testRestTemplate.exchange(getUrl() + "/get",
+                HttpMethod.POST,
+                entity,
+                String.class);
+
+        //THEN
+        assertThat(response.getBody(), equalTo("artist2"));
+    }
+
+    public List<Artist> getArtistList(){
```
[the remark this answers is not in the document]
Comment (phlpphcknr) [reply backend/src/test/java/com/hackner/musicompass/controller/RecommendationControllerTest.java:142]: thx. changed.

File: backend/src/test/java/com/hackner/musicompass/service/RecommendationServiceTest.java (hunk 1 above)
Comment (phlpphcknr) [reply backend/src/test/java/com/hackner/musicompass/service/RecommendationServiceTest.java]: fixed, due to proper error handling

File: frontend/src/components/GlobalStyle.js (hunk 2 above)
Comment (phlpphcknr) [reply frontend/src/components/GlobalStyle.js:17]: thx. done

File: backend/src/test/java/com/hackner/musicompass/service/RecommendationServiceTest.java [hunk 4]
```diff
@@ -141,7 +143,96 @@ public void setRecommendationTags(){
         verify(artistMongoDb).save(artistAfter);
         assertThat(actual, equalTo(recommendationTagsAfter));
     }
-/*
+
+    @Test
+    @DisplayName("Request Artist Recommendation with tag combination which is not available")
```
Comment (jamarob) [inline backend/src/test/java/com/hackner/musicompass/service/RecommendationServiceTest.java:148]: update this message as well ;)

File: frontend/src/components/GetArtistRecommendation.js [hunk 5]
```diff
@@ -22,7 +22,10 @@ export default function GetArtistRecommendation(){
     function onClick() {
         getRecommendation({genreTags, rolesTags, genderTag})
             .then((response) => history.push(`/artist/${response}`))
-            .catch((error) => console.error(error))
+            .catch((error) => {
+                console.error(error)
```
Comment (jamarob) [inline frontend/src/components/GetArtistRecommendation.js:26]: log not needed, imho

**Review by jamarob — APPROVED** [review]
Excellent! 🎖️  Don't forget to update the display message of your tests.

File: backend/src/test/java/com/hackner/musicompass/service/RecommendationServiceTest.java (hunk 4 above)
Comment (phlpphcknr) [reply backend/src/test/java/com/hackner/musicompass/service/RecommendationServiceTest.java]: done

File: frontend/src/components/GetArtistRecommendation.js (hunk 5 above)
Comment (phlpphcknr) [reply frontend/src/components/GetArtistRecommendation.js]: I am not using it now. deleted

**Pull request merged**